index.html

```diff
@@ -352,132 +348,145 @@
         </dt>
         <dd>
           These events are represented by <a>VideoProcessorEvent</a> objects
-          that are dispatched at a <a>processing video worker</a> and provide
-          read access to <a>input frame</a> data and write access to <a>output
+          that are dispatched at a <a>VideoProcessor</a> and provide read
+          access to <a>input frame</a> data and write access to <a>output
           frame</a> data.
         </dd>
       </dl>
     </section>
     <section>
       <h2>
-        Worker extensions
+        <a>VideoMonitor</a> and <a>VideoProcessor</a> interfaces
       </h2>
-      <section>
-        <h2>
-          <code>DedicatedWorkerGlobalScope</code> interface
-        </h2>
-        <p>
-          To enable generating, processing, and analyzing of video data
-          directly using JavaScript in a worker, the
-          <a>DedicatedWorkerGlobalScope</a> interface is extended with an
-          <a>onvideomonitor</a> and <a>onvideoprocess</a> <a>event handler IDL
-          attribute</a>. <a data-lt="video monitor event">Video monitor
-          events</a> and <a data-lt="video processor event">video processor
-          events</a> are dispatched at the <a>DedicatedWorkerGlobalScope</a>
-          object.
-        </p>
-        <pre class="idl">
-          partial interface DedicatedWorkerGlobalScope {
-                          attribute EventHandler onvideomonitor;
-                          attribute EventHandler onvideoprocess;
+      <p>
+        The <a>VideoMonitor</a> interface is used to analyze and
+        <a>VideoProcessor</a> interface in addition to generate and process
+        video data directly using JavaScript. The <a data-lt=
+        "video monitor event">video monitor events</a> are dispatched at
+        <a>VideoMonitor</a> objects, and <a data-lt=
+        "video processor event">video processor events</a> are dispatched at
+        <a>VideoProcessor</a> objects.
+      </p>
+      <pre class="idl">
+          [Constructor]
```

Comment:
The constructor should accept a script url and should expose to both Window and Worker, right?